fair/src/autogluon/fair/learners/fair.py

```diff
@@ -0,0 +1,635 @@
+         only the metrics of the original predictor.
+         parameters
+        ----------
+        data: (optional) a pandas dataframe to evaluate over. If not provided evaluate over
+                the dataset provided at initialisation.
+        groups (optional) a specification of the groups (see class defintion). If not provided use
+                the defintion provided at init.
+        metrics: (optional) a dictionary where the keys are metric names and the elements are either
+                    scoreables or group metrics. If not provided report the standard metrics
+                    reported by SageMaker Clarify
+                    https://mkai.org/learn-how-amazon-sagemaker-clarify-helps-detect-bias
+        returns
+        -------
+        a pandas dataset containing rows indexed by fairness measure name, and columns by
+        ['original', 'updated']
+         """
+        if metrics is None:
+            metrics = group_metrics.clarify_metrics
+
+        if data is None:
+            data = self.validation_data
+            labels = self.y_true
+            y_pred_proba = self.predictor.predict_proba(data)
+        else:
+            labels = np.asarray(data[self.predictor.label])
+            y_pred_proba = self.predictor.predict_proba(data)
+            labels = (labels == self.predictor.positive_class) * 1
+        groups = self.groups_to_numpy(groups, data)
+
+        collect = self.fairness_metrics(labels, y_pred_proba, groups, metrics)
+        collect.columns = ['original']
+
+        if np.any(self.offset):
+            y_pred_proba = self.predict_proba(data)
+            new_pd = self.fairness_metrics(labels, y_pred_proba, groups, metrics)
+            new_pd.columns = ['updated']
+            collect = pd.concat([collect, new_pd], axis='columns')
+        return collect
+
+    def fairness_metrics(self, y_true: np.ndarray, proba, groups: np.ndarray,
+                         metrics: dict) -> pd.DataFrame:
+        """Helper function for evaluate_fairness
+        Report fairness metrics that do not require additional information.
+        parameters
+        ----------
+        y_true: numpy array containing true binary labels of the dataset
+        proba: numpy or pandas array containing the output of predict_proba
+        groups: numpy array containing discrete group labelling
+        metrics: a dictionary where keys are the names and values are either
+        Scorable or a BaseGroupMetric.
+        returns
+        -------
+        a pandas dataframe of fairness metrics
+        """
+        values = np.zeros(len(metrics))
+        names = []
+        for i, k in enumerate(metrics.keys()):
+            names.append(k)
+            values[i] = dispatch_metric(metrics[k], y_true, proba, groups)
+
+        return pd.DataFrame(values, index=names)
+
+    def evaluate_groups(self, data=None, groups=None, metrics=None, return_original=False):
```

Comment:
Can we highlight the max and the min in each column?  This would require a separate rendering, since this doesn't make sense in the dataframe.  For example, we could have the dataframe outputted, and then if `render=True` or something then if renders the dataframe with `Markdown` from `IPython.display`.  More generally, we could make the output prettier using this library.